exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Database.kt

```diff
@@ -72,6 +72,17 @@ class Database private constructor(
         return this
     }
 
+    /** Whether [Database.connect] was invoked with a [DataSource] argument. */
+    internal var connectsViaDataSource = false
+        private set
+
+    /**
+     * The transaction isolation level defined by a [DataSource] connection.
+     *
+     * This should only hold a non-null value if [connectsViaDataSource] has been set to `true`.
+     */
+    internal var dataSourceIsolationLevel: Int? = null
```

Comment:
I would try to avoid nullable fields because of boxing, but it shouldn't be a problem here